python/paddle/distributed/fleet/launch.py

```diff
@@ -432,6 +459,78 @@ def launch_ps(args, distribute_mode):
     return
 
 
+def launch_ipu(args):
+    # The number of replicas for data parallel
+    assert (args.num_ipus % args.ipus_per_replica) == 0, \
+                "The number of IPUs:{} mod the number of IPUs per replica:{} must == 0".format(args.num_ipus, args.ipus_per_replica)
+    num_replicas = int(args.num_ipus / args.ipus_per_replica)
+
+    # The number of processes
+    nproc_per_node = 1
+    if args.nproc_per_node is not None:
+        nproc_per_node = args.nproc_per_node
```

Comment:
What is the difference between `nproc_per_node` and `ipus_per_replica`?

Reply:
For poprun distributed computing, we have the following concepts:
1. ipus_per_replica: the number of IPUs per instance.

2. replica:  data parallel working in a process. For example: Require 2 IPUs (ipus_per_replica=2) per instance, and set replicas=2 (2 instances), then PopART will totally acquire 4 IPUs per process for data parallel.

3. process: each process is able to include multi instances (replicas > 1)

4. hosts (ips) : each host is able to include multi processes.

`nproc_per_node`: The number of processes per host.
`ipus_per_replica`: The number of IPUs per instance (per replica).